File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;

/**
 * 品牌接口
 * 
 * @author yoofale
 *
 */
public interface BrandService {
	public List<TbBrand> findAll();

	/**
	 * 品牌分页
	 * 
	 * @param pageNum当前页码
	 * @param pageSize每页记录数
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);

	/**
	 * 查询分页(重载)
	 * 
	 * @param brand
	 *            要查询的条件
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	public PageResult findPage(TbBrand brand, int pageNum, int pageSize);

	/**
	 * 增加品牌
	 */
	public void add(TbBrand brand);

	/**
	 * 查询到要修改的对象
	 * 
	 * @param id
	 * @return
	 */
	public TbBrand findOne(Long id);

	/**
	 * 将修改后的结果存回去
	 * 
	 * @param brand
	 */
	public void update(TbBrand brand);

	/**
	 * 删除
	 * 
	 * @param ids
	 */
	public void delete(long[] ids);
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService',function($http){
			//查找所有
			this.findAll=function(){
				return $http.get('../brand/findAll.do');
			}
			//分页查找
			this.findPage=function(page,size){
				return $http.get('../brand/findPage.do?page='+page +'&size='+size);
			}
			//查找实体
			this.findOne=function(id){
				return $http.get('../brand/findOne.do?id='+id);
			}
			//添加
			this.add=function(entity){
				return $http.post('../brand/add.do',entity);
			}
			//修改
			this.update=function(entity){
				return $http.post('../brand/uodate.do',entity);
			}
			//删除
			this.dele=function(ids){
				return $http.get('../brand/delete.do?ids='+ids);
			}
			//条件查询
			this.search=function(page,size,searchEntity){
				return $http.post('../brand/search.do?page='+page +'&size='+size,searchEntity);
			}
			//下拉列表数据
			this.selectOptionList=function(){
				return $http.get('../brand/selectOptionList.do')
			}
		});